oof_util.php

```diff
@@ -337,20 +337,25 @@ public static function add_oof_on_send($args)
         $now = new DateTime();
         
         $user_prefs = $rcmail->user->get_prefs();
-        $add_structured_data = $user_prefs['include_structured_data_in_email'];
-        $days_prior = $user_prefs['days_prior_include_structured_data_in_email'];
-        $now_plus_days_prior = (new DateTime())->modify("+$days_prior days");
+        $add_structured_data_in_email = $user_prefs['include_structured_data_in_email'];
+        $days_prior = 0;
+
+        // Check if we have a integer
+        if(intval(isset($user_prefs['days_prior_include_structured_data_in_email']))!= 0)
+            $days_prior = $user_prefs['days_prior_include_structured_data_in_email'];
+
+        // Calculating when to start sending or replying messages
+        $now_minus_days_prior = (new DateTime())->modify("- {$days_prior} day");
 
         /*
          * Add OOF structured data iff the user prefs allow us and
```

Comment:
The value days_prior comes from the user_pref field, (that gets it from the gui object which returns empty strings) this value can be also an empty string. Thats why this condition was often false when it shouldnt.